lib/nostr.js

```diff
@@ -25,3 +32,93 @@ export function nostrZapDetails (zap) {
 
   return { npub, content, note }
 }
+
+async function publishNostrEvent(signedEvent, relay) {
+  return new Promise((resolve, reject) => {
+    const timeout = 1000;
+    const wsRelay = new WebSocket(relay);
+    let timer;
+    let isMessageSentSuccessfully = false;
+
+    function timedout() {
+      clearTimeout(timer);
+      wsRelay.close();
+      reject(new Error(`relay timeout for ${relay}`));
+    }
+
+    timer = setTimeout(timedout, timeout);
+
+    wsRelay.onopen = function () {
+      clearTimeout(timer);
+      timer = setTimeout(timedout, timeout);
+      wsRelay.send(JSON.stringify(['EVENT', signedEvent]));
+    };
+
+    wsRelay.onmessage = function (msg) {
+      const m = JSON.parse(msg.data);
+      if (m[0] === 'OK') {
+        isMessageSentSuccessfully = true;
+        clearTimeout(timer);
+        wsRelay.close();
+        console.log('Successfully sent event to', relay);
+        resolve();
+      }
+    };
+
+    wsRelay.onerror = function (error) {
+      clearTimeout(timer);
+      reject(new Error(`relay error: Failed to send to ${relay}`));
+    };
+
+    wsRelay.onclose = function () {
+      clearTimeout(timer);
+      if (!isMessageSentSuccessfully) {
+        reject(new Error(`relay error: Failed to send to ${relay}`));
+      }
+    };
+  });
+}
+
+export async function crosspostDiscussion(item, id, relays) {
+  try {
+    const userPubkey = await window.nostr.getPublicKey();
+    const timestamp = Math.floor(Date.now() / 1000);
+
+    const event = {
+      created_at: timestamp,
+      kind: 30023,
+      content: item.text,
+      tags: [
+        ['d', `https://stacker.news/items/${id}`],
+        ['a', `30023:${userPubkey}:https://stacker.news/items/${id}`, 'wss://nostr.mutinywallet.com'],
```

Comment:
This `a` tag should probably list out the relays from the function parameter, right?